### DISCUSSION TITLE
Production mode

### DISCUSSION THREAD

**Discussion by yazan-alnugnugh** [spatie/laravel-markdown#31] [Q&A]
when use package on locale server is working perfectly but when I work on a production server is not working The code appears like that

before : 

```
# My title

This is a [link to our website](https://spatie.be)

```php
echo 'Hello world';
```

```

after : 

My title
This is a [link to our website](https://spatie.be) ``` php echo 'Hello world'; ```

**Comment by robfrancken** [comment]
As far as I can tell, the shiki node_module needs to be installed on your server.

Try executing:

```php
$highlighted = \Spatie\ShikiPhp\Shiki::highlight(
    code: '<?php echo "Hello World"; ?>',
    language: 'php',
    theme: 'github-light',
);
```

If you get: `Error: Cannot find module 'shiki'`, `laravel-markdown` can't find your node module.

This was my problem.

After doing `npm install --production`, it worked.

**Reply by hilmanski** [reply]
can you verified , we should run that in production ?

**Reply by robfrancken** [reply]
From the docs: [link](https://docs.npmjs.com/cli/v8/commands/npm-install)

> With the --production flag (or when the NODE_ENV environment variable is set to production), npm will not install modules listed in devDependencies.

Having said that, I actually went a different route and used the Github API to parse markdown: [link](https://calebporzio.com/parsing-markdown-the-easy-way-with-code-highlighting)

**Reply by hilmanski** [reply]
Thanks for the refrence, here is what works for me https://laracasts.com/discuss/channels/laravel/spatie-laravel-markdown-not-working-locally
